### PR TITLE
Add lastOpenedAt and JoinedAt for communities

### PR DESCRIPTION
**Pull request by ibrkhalil**
Required for https://github.com/status-im/status-mobile/issues/17337

### Summary

Adds last-opened-at and joined-at for communities to help with their sorting
QA notes: This won't require QA review as the PR that's going to use these attributes is still WIP

status: ready

### CODE REVIEW

File: src/status_im/contexts/communities/events.cljs
Comment (Parveshdhull) [inline src/status_im/contexts/communities/events.cljs]: Please add an empty line here

File: src/status_im/contexts/communities/overview/view.cljs
Comment (Parveshdhull) [inline src/status_im/contexts/communities/overview/view.cljs]: This should be part of navigation event which is opening community. (we are adding an event in https://github.com/status-im/status-mobile/pull/18465)

File: src/status_im/contexts/communities/events.cljs
Comment (Parveshdhull) [reply src/status_im/contexts/communities/events.cljs]: also please can you elaborate, how this change related to current pr?

File: src/status_im/contexts/communities/events.cljs
Comment (Parveshdhull) [inline src/status_im/contexts/communities/events.cljs]: what we are handling here? Do we need this?

Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: After returning the response of the community with the update LastOpenedAt value, We need to update the community in app-db for the changes to happen in the UI.

Comment (Parveshdhull) [reply src/status_im/contexts/communities/events.cljs]: Please can you elaborate, what changes will be updating in UI after updating LastOpenedAt?

File: src/status_im/contexts/communities/events.cljs
Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: Sorry messed thing up when solving conflicts, Reverting

File: src/status_im/contexts/communities/events.cljs
Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: The sorting of the communities, That'll be done in https://github.com/status-im/status-mobile/issues/17337

File: src/status_im/contexts/communities/overview/view.cljs
Comment (ibrkhalil) [reply src/status_im/contexts/communities/overview/view.cljs]: Okay, I'll merge it after that one is merged and the event dispatching there.

**Comment by ibrkhalil** [comment]
Note: Only merge after https://github.com/status-im/status-mobile/pull/18465 and adding the event dispatching to the community/navigate-to-community event
`[:communities/update-last-opened-at id] --> Should only happen in [:communities/navigate-to-community-overview] and [:communities/navigate-to-community-chat]`

File: src/status_im/contexts/communities/overview/view.cljs
Comment (ilmotta) [reply src/status_im/contexts/communities/overview/view.cljs]: Also the `id` of a community never changes, so there's no need to make the dispatch code reactive with `use-effect`.

File: src/status_im/contexts/communities/events.cljs
Comment (ilmotta) [reply src/status_im/contexts/communities/events.cljs]: @ibrkhalil, what we're doing here is very inefficient. Every time the community overview screen is opened we:

1. Update the timestamp on the backend.
2. Recursively transform a gigantic `Community` JS payload from scratch to a Clojure data structure.
3. If the community has token permissions (which many will), we re-dispatch other events, such as `:communities/check-permissions-to-join-community`, which may cause yet more db updates.

But why would we want to reprocess community permissions if the originating request just wanted to update a single timestamp column in the DB?

If we want to prioritize performance, we would make the event you wrote in this PR `:communities/update-last-opened-at` do a quick `assoc-in` in the app db and that's it. Much much cheaper.

Something along these lines:

```clojure
(rf/reg-event-fx :communities/update-last-opened-at
 (fn [_ [community-id]]
   {:json-rpc/call [{:method     "wakuext_communityUpdateLastOpenedAt"
                     :params     [community-id]
                     :on-success [:communities/update-last-opened-at-success]
                     :on-error   ...}]}))

(rf/reg-event-fx :communities/update-last-opened-at-success
 (fn [{:keys [db]} [^js community]]
   (let [community-id   (oops/oget community :id)
         last-opened-at (oops/oget community :lastOpenedAt)]
     {:db (assoc-in db [:communities community-id :last-opened-at] last-opened-at)})))
```

Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: What might be even better, If we could return just the value of the lastOpenedAt instead of returning the whole MessengerResponse, But I'd have to investigate if that's possible.

Comment (ilmotta) [reply src/status_im/contexts/communities/events.cljs]: Yes, that'd be even better 👍🏼 Thinking about the future, I would prefer if our clients could always request what data they need, similar to what's possible with GraphQL endpoints.

Comment (ibrkhalil) [reply src/status_im/contexts/communities/events.cljs]: Done

**Review by ilmotta — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

**Pull request merged**
